docs/Compiler-Hardening-Guides/Compiler-Annotations-for-C-and-C++.md

```diff
@@ -101,12 +102,16 @@ void my_free(void *ptr);
 void *my_malloc(size_t size) __attribute__ ((malloc, malloc (my_free, 1)));
 ~~~
 
-Note that to benefit both from the associated optimizations and improved detection of memory errors functions should be marked with _both_ the form of the attribute without arguments and the form of the attribute with one or two arguments. [[Extended example at Compiler Explorer](https://godbolt.org/z/bc97ahbnd)]
+Note that to benefit from both the associated `noalias` optimization and the improved detection of memory errors,
+functions should be marked with _both_ the zero-argument `malloc` attribute and the form of the attribute with one
```

Comment:
Nit: superflous linebreaks likely due to editor line wrap settings. To be consistent with the rest of the guide it'd be better to format the paragraph as a single line.